File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    if number < 1:
        raise ValueError("Number must be equal or greater than 1")
    if number == 1:
        return "deficient"
    result = 0
    div = 2
    factors = {1}
    while number != div:
        if number % div == 0:
            factors.add(div)
        div += 1
   
    result = sum(factors)
    if result == number:
        return "perfect"
    elif result > number:
        return "abundant"
    else:
        return "deficient"

File: go/perfect-numbers/perfect_numbers.go
// Package perfect provides Perfect Number solution.
package perfect

import "errors"

// Classification represents a number classification based on their aliquot sum.
type Classification string

const (
	ClassificationPerfect   = Classification("perfect")
	ClassificationAbundant  = Classification("abundant")
	ClassificationDeficient = Classification("deficient")
)

var ErrOnlyPositive = errors.New("n must be a positive number")

// Classify returns the number type based on their aliquot sum.
func Classify(n int64) (Classification, error) {
	if n < 1 {
		return "", ErrOnlyPositive
	}
	var aliquotSum int64
	for i := int64(1); i < n; i++ {
		if n%i == 0 {
			aliquotSum += i
		}
	}
	var result Classification
	switch {
	case aliquotSum == n:
		result = ClassificationPerfect
	case aliquotSum > n:
		result = ClassificationAbundant
	case aliquotSum < n:
		result = ClassificationDeficient
	}
	return result, nil
}
